src/libraries/System.Private.Xml/tests/XmlWriter/DisposeTests.cs

```diff
@@ -121,5 +123,155 @@ public static void XmlWriterDisposeWorksWithDerivingClasses()
+        public static async Task AsyncWriter_DisposeAsync_ShouldCall_FlushAsyncWriteAsyncOnly_StreamWriter()
+        {
+            using (var stream = new ForceAsyncSyncStream())
+            await using (var writer = XmlWriter.Create(stream, new XmlWriterSettings() { Async = true }))
+            {
+                await writer.WriteStartDocumentAsync();
+                await writer.WriteStartElementAsync(string.Empty, "root", null);
+                await writer.WriteStartElementAsync(null, "test", null);
+                await writer.WriteAttributeStringAsync(string.Empty, "abc", string.Empty, "1");
+                await writer.WriteEndElementAsync();
+                await writer.WriteEndElementAsync();
+            }
+            
+        }
+
+        [Fact]
+        public static async Task XmlWriter_AsyncSyncResult_ShouldBeSame_AfterDispose_StreamWriter()
+        {
+            using (var stream1 = new MemoryStream())
+            using (var stream2 = new MemoryStream())
+            using (var stream3 = new MemoryStream())
+            {
+                using (var asyncWriter = XmlWriter.Create(stream1, new XmlWriterSettings() { Async = true }))
+                {
+                    await asyncWriter.WriteStartDocumentAsync();
+                    await asyncWriter.WriteStartElementAsync(string.Empty, "root", null);
+                    await asyncWriter.WriteStartElementAsync(null, "test", null);
+                    await asyncWriter.WriteAttributeStringAsync(string.Empty, "abc", string.Empty, "1");
+                    await asyncWriter.WriteEndElementAsync();
+                    await asyncWriter.WriteEndElementAsync();
+                }
+
+                await using (var asyncWriter = XmlWriter.Create(stream2, new XmlWriterSettings() { Async = true }))
+                {
+                    await asyncWriter.WriteStartDocumentAsync();
+                    await asyncWriter.WriteStartElementAsync(string.Empty, "root", null);
+                    await asyncWriter.WriteStartElementAsync(null, "test", null);
+                    await asyncWriter.WriteAttributeStringAsync(string.Empty, "abc", string.Empty, "1");
+                    await asyncWriter.WriteEndElementAsync();
+                    await asyncWriter.WriteEndElementAsync();
+                }
+
+                using (var syncWriter = XmlWriter.Create(stream3, new XmlWriterSettings() { Async = false }))
+                {
+                    syncWriter.WriteStartDocument();
+                    syncWriter.WriteStartElement(string.Empty, "root", null);
+                    syncWriter.WriteStartElement(null, "test", null);
+                    syncWriter.WriteAttributeString(string.Empty, "abc", string.Empty, "1");
+                    syncWriter.WriteEndElement();
+                    syncWriter.WriteEndElement();
+                }
+
+                Assert.Equal(Encoding.UTF8.GetString(stream1.GetBuffer()), Encoding.UTF8.GetString(stream2.GetBuffer()));
```

Comment:
It would make sense to compare bytes separately as well (so you get nice message for string diff and any message for other diff). There might be a difference in encoding, i.e. one payload has a BOM and other doesn't.